launch/launch/logging/__init__.py

```diff
@@ -30,123 +30,203 @@
-       Defaults to regular log file handlers for logging if no factory is given.
-    """
-    if level is not None:
-        logging.root.setLevel(level)
-    if screen_format is not None:
-        if screen_format == 'default':
-            screen_format = '[{levelname}] [{name}]: {msg}'
-            if screen_style is not None:
-                raise ValueError(
-                    'Cannot set a custom format style for the "default" screen format.'
-                )
-        if screen_format == 'default_with_timestamp':
-            screen_format = '{created:.7f} [{levelname}] [{name}]: {msg}'
-            if screen_style is not None:
-                raise ValueError(
-                    'Cannot set a custom format style for the '
-                    '"default_with_timestamp" screen format.'
-                )
-        if screen_style is None:
-            screen_style = '{'
-        launch_config.screen_formatter = logging.Formatter(
-            screen_format, style=screen_style
-        )
-        if launch_config.screen_handler is not None:
-            launch_config.screen_handler.setFormatter(launch_config.screen_formatter)
-    if log_format is not None:
-        if log_format == 'default':
-            log_format = '{created:.7f} [{levelname}] [{name}]: {msg}'
-            if log_style is not None:
-                raise ValueError(
-                    'Cannot set a custom format style for the "default" log format.'
-                )
-        if log_style is None:
-            log_style = '{'
-        launch_config.file_formatter = logging.Formatter(
-            log_format, style=log_style
-        )
-        for handler in launch_config.file_handlers.values():
-            handler.setFormatter(launch_config.file_formatter)
-    if log_handler_factory is not None:
-        launch_config.log_handler_factory = log_handler_factory
-    if log_dir is not None:
-        if any(launch_config.file_handlers):
-            import warnings
-            warnings.warn((
-                'Loggers have been already configured to output to log files below {}. '
-                'Proceed at your own risk.'
-            ).format(launch_config.log_dir))
-        if not os.path.isdir(log_dir):
-            raise ValueError('{} is not a directory'.format(log_dir))
-        launch_config.log_dir = log_dir
+class LaunchConfig:
+    """Launch Logging Configuration class."""
+
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self._log_dir = None
+        self.file_handlers = {}
+        self.screen_handler = None
+        self.screen_formatter = None
+        self.file_formatter = None
+        self._log_handler_factory = None
+
+    def set_level(self, new_level):
+        """
+        Set up launch logging verbosity level for all loggers.
+
+        :param new_level: the default log level used for all loggers.
+        """
+        logging.root.setLevel(new_level)
+
+    level = property(None, set_level)
```

Comment:
@clalancette nit: you may use decorators instead.

Reply:
Actually, not for this particular one.  You can only use decorators if you implement a getter, apparently.  I learned that while doing this :).  So if you just want a setter, you have to do it this way.  Alternatively, you could argue that we want a getter for `level`, in which case I could switch to decorators (here and elsewhere).

Reply:
I ended up adding a getter too, so now I've switch to properties.